src/server/hll_family.cc

```diff
@@ -281,14 +284,22 @@ void PFMerge(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+}  // namespace
+namespace {
+namespace Acl {
+namespace Cat = AclCategory;
+constexpr uint32_t kPFAdd = Cat::WRITE | Cat::HYPERLOGLOG | Cat::FAST;
```

Comment:
Maybe introduce a macro to reduce the boilerplate? i.e. `ACL_CATEGORY(WRITE, FAST, HYPERLOGLOG)`

Reply:
This will change the syntax but not really reduce the boilerplate. I really don't like macros (unless there is a very good reason to use them) and since this won't add much here I rather not do it atm.

Reply:
Sure! most comments I had - are purely subjective. Regarding macros - surprisingly I actually find them very useful for the production-grade code (i.e debug checks etc). (this case is irrelevant of course). Macros are one of the bigger reasons why I do not enjoy coding in golang or Java. Though I must say, the Rust "macro" language is 10 times better than C/C++.

Reply:
> Regarding macros - surprisingly I actually find them very useful for the production-grade code (i.e debug checks etc)

100% macros have good use cases and that's why are a necessary evil :D IMO, I have seen a lot of macro wonkery over the years (in places where templates would have been a better option) and my despise towards them comes from that. 

p.s. one of the cases that macros are infamous but very very useful, is when you want to implement `reflection` in C++ with templates -- that is a library feature not a language feature. You need macros for that (since we don't have reflection implemented as a language feature -- soon though hopefully in 26 or 29) 

> for java and golang 

Not a big fun of GC based languages especially when you want raw control. If you want to fine tune something that relies on GC you pretty much screwed -- meaning that you end up fighting the GC instead of solving the actual issue 

> Though I must say, the Rust "macro" language is 10 times better

I never tried Rust macros. I played with rust a little bit but IMO I don't buy its safety guarantees mostly because:
1. One way or another you will write rust that interferes with C/C++ code. You need `unsafe` for that so it kinda beats the purpose of the language
2. Compile times are SLOW AF. I have seen small codebases requiring hours to compile. People who complain about TMP and SFINAE compilation times would `panic` (pan intended here :rofl: ) when they see the compile times of Rust in big codebases.
3. You want to leverage RUST without a performance hit? You still need to write `unsafe` code :/
4. There are some TMP tricks that you just can't do with rust no matter how you try
5. The quality of the libraries found in rust are subpar compared to C++. One example is `p2300` aka senders and receivers. It's a very complicated async model but it has `0` overhead compared to other async abstractions like `futures and promises`. The former does not require dynamic memory allocation neither syncronization primitives for attaching continuations while the later does. You won't find high calibre libraries like that in Rust... Yes there is a learning curve here which is *expensive* but very rightfully so for certain cases and problems.

Plz don't get me wrong, I think every language and feature has a tool -- I would program in any of them if it solves my problem elegantly. In my view, it's always a matter of asking the correct questions when you are implementing something (a good example here is your decision with boost fibers and helio -- can they be optimized further ? YES! Is it worth it ? NO! Why? Because there are other parts that bring more value!)

Reply:
Btw I am not ranting here, I just love to think out loud and discuss these kind of things. I can do that for hours. Save it for when I am in Israel. I WOULD LOOOOOVE to discuss this!

Reply:
Aaaaaalso one more thing. The goto async abstraction for Rust is https://tokio.rs/ which is *AWESOME* to play with and implement/try out new things. It just works! 

However, I could write a lib that outperforms tokio (both in terms of performance and memory) *any* day :smile: